src/eventHandling.js

```diff
@@ -40,33 +40,67 @@ function sayCorrectOrIncorrect(span, correct) {
 }
 
 function handleSpace() {
-  // check the text has even been loaded
-  if (comparisonWord !== null) {
+  // checks the text has been loaded, and the game has not finished
+  if (comparisonWord !== null && gameOver == false) {
     const inputTextbox = document.querySelector('#input-textbox')
 
-    currentWordSpan.classList.remove("current-word")
-
+    if (currentWordSpan) {
+      currentWordSpan.classList.remove("current-word")
+    }
     const correct = comparisonWord === inputTextbox.value
 
     inputTextbox.value = ''
 
-    if (correct) {
+    if (correct && currentWordSpan) {
       currentWordSpan.classList.add("correct")
-    } else {
+      totalCorrectWords++
+    } else if (currentWordSpan) {
       currentWordSpan.classList.add("incorrect")
+      totalIncorrectWords++
+    }
+
+    if (currentWordSpan) {
+      cleanupTemporaryScreenReaderText(currentWordSpan)
+
+      // get the next word
+      currentWordSpan = wordSpanArray.shift()
+      comparisonWord = comparisonWordsArray.shift()
```

Comment:
wrt earlier comment in this file, if you add `?? null` to the end of these two lines, you don't need all of the `if (currentWordSpan)`

Reply:
as then at the end `comparisonWord === null`

Reply:
I knew there was a better way.... thanks for figuring it out

Reply:
ah yep, I suppose you'd need that last `if (currentWordSpan)` in the file, as it's trying to do stuff with the next word that doesn't exist

Reply:
*in the function

Reply:
done - yes the final `if (currentWordSpan) ` is needed but the others aren't when the ?? null is there